UN-OCHA/hpc-cdm
# HPC-9975 Remove `use-query-params`

## Nature of PR

- Bug-fix
- Hot-fix
- 🟢 Feature
- Testing
- 🟢 Package update
- CI


## Description

This PR is related to HPC-9965, read for more context.

In this PR we focus on removing `use-query-params` package. It became obsolete since with the use of `react-router` data API hooks (precisely, `useSearchParams()`) and using `io-ts` to write runtime validation, we can achieve the same objective.

We do this in order to reduce dependency of external libraries.

## How to test changes

To test changes I would try to manipulate the URL to see how it reacts even to unexpected `params` and with wrong data on it.

Beside this, I would focus mainly on reading logic for `useQueryParams()` to verify it really covers corner cases
